### PR TITLE
36634-Sign in / Sign up

### PR DESCRIPTION
**Pull request by luongvietdung**

### CODE REVIEW

File: app/views/devise/mailer/confirmation_instructions.html.erb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<p>Welcome <%= @email %>!</p>
+
+<p>You can confirm your account email through the link below:</p>
```
Comment (vinhnd2010) [inline app/views/devise/mailer/confirmation_instructions.html.erb]: User i18n. :)

File: app/views/devise/mailer/reset_password_instructions.html.erb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<p>Hello <%= @resource.email %>!</p>
+
+<p>Someone has requested a link to change your password. You can do this through the link below.</p>
+
+<p><%= link_to 'Change my password', edit_password_url(@resource, reset_password_token: @token) %></p>
+
+<p>If you didn't request this, please ignore this email.</p>
+<p>Your password won't change until you access the link above and create a new one.</p>
```
Comment (vinhnd2010) [inline app/views/devise/mailer/reset_password_instructions.html.erb]: i18n.

File: app/views/devise/mailer/unlock_instructions.html.erb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<p>Hello <%= @resource.email %>!</p>
+
+<p>Your account has been locked due to an excessive number of unsuccessful sign in attempts.</p>
+
+<p>Click the link below to unlock your account:</p>
+
+<p><%= link_to 'Unlock my account', unlock_url(@resource, unlock_token: @token) %></p>
```
Comment (vinhnd2010) [inline app/views/devise/mailer/unlock_instructions.html.erb]: i18n

File: app/assets/stylesheets/layouts.scss [hunk 4]
```diff
@@ -47,3 +47,41 @@ h1, h2, h3, h4, h5 {
...
+  border-bottom-left-radius: 0;
+  border-bottom-right-radius: 0;
+}
+
+input[type="password"] {
+  border-top-left-radius: 0;
+  border-top-right-radius: 0;
+}
+
+
```
Comment (ThuBM) [inline app/assets/stylesheets/layouts.scss]: remove

File: app/views/devise/sessions/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+      <%= f.email_field :email, autofocus: true,
+        placeholder: t("devise.sessions.new.email"), class: "form-control" %>
+    </div>
+    <div class="form-group">
+      <%= f.password_field :password,
+        placeholder: t("devise.sessions.new.password"), autocomplete: "off",
+        class: "form-control" %>
+    </div>
+
+
```
Comment (ThuBM) [inline app/views/devise/sessions/new.html.erb]: remove 2 blank lines

File: app/views/devise/shared/_links.html.erb [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<% if controller_name != 'sessions' %>
```
Comment (ThuBM) [inline app/views/devise/shared/_links.html.erb]: change '  => "

File: Gemfile [hunk 7]
```diff
@@ -22,6 +22,7 @@ gem "jbuilder", "~> 2.0"
 # bundle exec rake doc:rails generates the API under doc/api.
 gem "sdoc", "~> 0.4.0", group: :doc
 
+gem 'devise'
```
Comment (trungnn142) [inline Gemfile]: change ' => "

File: app/views/devise/registrations/new.html.erb [hunk 8]
```diff
@@ -0,0 +1,30 @@
+<div class="wrapper">
```
Comment (ThuBM) [inline app/views/devise/registrations/new.html.erb]: no need this file

File: app/helpers/devise_helper.rb [hunk 9]
```diff
@@ -0,0 +1,18 @@
+module DeviseHelper
```
Comment (ThuBM) [inline app/helpers/devise_helper.rb]: no need this file

File: app/views/devise/registrations/new.html.erb [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+  <%= form_for(resource, as: resource_name, url: registration_path(resource_name),
+    html: {class: "form-signin"}) do |f| %>
+    <h3 class="form-signin-heading"><%= t "devise.registers.new.title" %></h3>
+    <%= devise_error_messages! %>
```
Comment (ThuBM) [inline app/views/devise/registrations/new.html.erb]: remove

File: config/initializers/devise.rb [hunk 11]
```diff
@@ -0,0 +1,265 @@
+# Use this hook to configure devise mailer, warden hooks and so forth.
```
Comment (ThuBM) [inline config/initializers/devise.rb]: remove all commented lines in this files

File: app/views/devise/shared/_links.html.erb [hunk 12]
```diff
@@ -0,0 +1,11 @@
+<% if controller_name != "sessions" %>
```
Comment (HieuMeo) [inline app/views/devise/shared/_links.html.erb]: Use unless instead of if

File: config/initializers/devise.rb [hunk 13]
```diff
@@ -0,0 +1,15 @@
+Devise.setup do |config|
+  config.mailer_sender = 'please-change-me-at-config-initializers-devise@example.com'
+
+  require 'devise/orm/active_record'
```
Comment (HieuMeo) [inline config/initializers/devise.rb]: Use "" instead of ''

File: app/views/layouts/application.html.erb [hunk 14]
```diff
@@ -1,17 +1,17 @@
...
-<body>
-  <%= render "layouts/header" %>
-  <div class="container" id="main">
-    <%= render "layouts/flash" %>
-    <%= yield %>
-  </div>
-  <%= render "layouts/footer" %>
-</body>
+  <head>
+    <title><%= full_title yield(:title) %></title>
```
Comment (HieuMeo) [inline app/views/layouts/application.html.erb]: Remove ()

File: app/views/devise/registrations/new.html.erb [hunk 15]
```diff
@@ -0,0 +1,30 @@
+<div class="wrapper">
+  <%= form_for(resource, as: resource_name, url: registration_path(resource_name),
+    html: {class: "form-signin"}) do |f| %>
+    <h3 class="form-signin-heading"><%= t "devise.registers.new.title" %></h3>
+    <%= render "shared/form_errors", object: f.object %>
+
+    <div class="form-group">
+        <%= f.text_field :name, autofocus: true,
```
Comment (dieunb) [inline app/views/devise/registrations/new.html.erb]: indent

File: app/views/devise/registrations/new.html.erb [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+          placeholder: t("devise.registers.new.name"), class: "form-control" %>
+    </div>
+    <div class="form-group">
+        <%= f.email_field :email, autofocus: true,
```
Comment (dieunb) [inline app/views/devise/registrations/new.html.erb]: indent

**Comment by dieunb** [comment]
OK

**Pull request merged**